components/storage/blockdevice/COMPONENT_OSPIF/OSPIFBlockDevice.h

```diff
@@ -0,0 +1,450 @@
+
+    // Send Generic command_transfer command to Driver
+    ospi_status_t _ospi_send_general_command(mbed::ospi_inst_t instruction_int, mbed::bd_addr_t addr, const char *tx_buffer,
+                                             mbed::bd_size_t tx_length, const char *rx_buffer, mbed::bd_size_t rx_length);
+
+    // Send command to read from the SFDP table
+    int _ospi_send_read_sfdp_command(mbed::bd_addr_t addr, void *rx_buffer, mbed::bd_size_t rx_length);
+
+    // Read the contents of status registers 1 and 2 into a buffer (buffer must have a length of 2)
+    ospi_status_t _ospi_read_status_registers(uint8_t *reg_buffer);
+
+    // Set the contents of status registers 1 and 2 from a buffer (buffer must have a length of 2)
+    ospi_status_t _ospi_write_status_registers(uint8_t *reg_buffer);
+
+    // Send set_frequency command to Driver
+    ospi_status_t _ospi_set_frequency(int freq);
+
+    // Update the 4-byte addressing extension register with the MSB of the address if it is in use
+    ospi_status_t _ospi_update_4byte_ext_addr_reg(bd_addr_t addr);
+
+    /*********************************/
+    /* Flash Configuration Functions */
+    /*********************************/
+    // Clear the device's block protection
+    int _clear_block_protection();
+
+    // Configure Write Enable in Status Register
+    int _set_write_enable();
+
+    // Wait on status register until write not-in-progress
+    bool _is_mem_ready();
+
+    // Enable Fast Mode - for flash chips with low power default
+    int _enable_fast_mode();
+
+    // Enable OPI Mode
+    int _enable_opi_mdoe();
+
+    // Query vendor ID and handle special behavior that isn't covered by SFDP data
+    int _handle_vendor_quirks();
+
+    /****************************************/
+    /* SFDP Detection and Parsing Functions */
+    /****************************************/
+    // Parse and Detect required Basic Parameters from Table
+    int _sfdp_parse_basic_param_table(mbed::Callback<int(mbed::bd_addr_t, void *, mbed::bd_size_t)> sfdp_reader,
+                                      mbed::sfdp_hdr_info &sfdp_info);
+
+    // Parse and Detect 4-Byte Address Instruction Parameters from Table
+    int _sfdp_parse_4_byte_inst_table(mbed::Callback<int(mbed::bd_addr_t, void *, mbed::bd_size_t)> sfdp_reader,
+                                      mbed::sfdp_hdr_info &sfdp_info);
+
+    // Detect the soft reset protocol and reset - returns error if soft reset is not supported
+    int _sfdp_detect_reset_protocol_and_reset(uint8_t *basic_param_table_ptr);
+
+    // Detect fastest read Bus mode supported by device
+    int _sfdp_detect_best_bus_read_mode(uint8_t *basic_param_table_ptr, int basic_param_table_size,
+                                        bool &set_quad_enable, bool &is_qpi_mode, bool &is_opi_mode);
+
+    // Enable Quad mode if supported (1-1-4, 1-4-4, 4-4-4 bus modes)
+    int _sfdp_set_quad_enabled(uint8_t *basic_param_table_ptr);
+
+    // Enable QPI mode (4-4-4)
```

Comment:
Please make a search for _qpi_ in this file to find all occurrences and replace with _opi_ where applicable. To avoid confusion. Please do the same for the c-file.